src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ReplaceLockStoreDsnByActualDefinitionPass.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\Lock\Store\StoreFactory;
+
+/**
+ * Replaces aliases passed to lock store factory with actual service definitions.
+ *
+ * @author Jack Thomas <jack.thomas@solidalpha.com>
+ */
+class ReplaceLockStoreDsnByActualDefinitionPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->getDefinitions() as $id => $definition) {
```

Comment:
The clean way would be rely on a tag to find the right definitions instead of inspecting all their factories.

But it would be even better to create the Reference object directly in FrameworkExtension at the time of processing the config.

Reply:
I think I tried that the first time and the service was not yet available. So I moved into doing a compiler pass.

Also. Where do service arguments '@sum.service' get converted into references? I was trying to find that code so I can call that instead of rolling my own, but I had a bit of trouble finding it.

Reply:
done